src/Common/src/System/Net/Internals/SocketExceptionFactory.cs

```diff
@@ -6,12 +6,41 @@
 
 namespace System.Net.Internals
 {
-    internal class SocketExceptionFactory
+    internal static partial class SocketExceptionFactory
     {
+        private class ExtendedSocketException : SocketException
+        {
+            private readonly EndPoint _endPoint;
+
+            public ExtendedSocketException(int errorCode, EndPoint endPoint)
+                : base(errorCode)
+            {
+                _endPoint = endPoint;
+            }
+
+            public ExtendedSocketException(SocketError socketError, int platformError)
+                : base((int)socketError)
+            {
+                HResult = platformError;
+            }
+
+            public override string Message
+            {
+                get
+                {
+                    if (_endPoint != null)
+                    {
+                        return base.Message + " " + _endPoint.ToString();
```

Comment:
This seems to create a different message compared with the original. Apps that took dependencies on the messages will have issues with this change.
I propose leaving this as is and changing it only if there are app-compat issues with apps that are getting ported..